maven-artifact/src/main/java/org/apache/maven/artifact/ArtifactUtils.java

```diff
@@ -89,10 +88,15 @@ public static String key(String groupId, String artifactId, String version) {
     }
 
     private static void notBlank(String str, String message) {
-        int c = str != null && str.length() > 0 ? str.charAt(0) : 0;
-        if ((c < '0' || c > '9') && (c < 'a' || c > 'z')) {
-            Validate.notBlank(str, message);
+        final int strLen = str != null ? str.length() : 0;
```

Comment:
str != null && !str.isEmpty() 

seems a little more straight-forward

Reply:
I don't understand, you mean:
```
final int strLen = str != null && !str.isEmpty() ? str.length() : 0;
```
I kinda fail to see how adding an addition condition makes things more straightforward, but I guess that's a matter of taste, as a lot of things when we go to that level.

Reply:
I was simply thinking

```
if (str != null && !str.isEmpty())
```

but this works too. Not a big deal.

Reply:
The point is that the length is used after, so we need it anyway.  I think the current version is the optimal one.